### PR TITLE
fix response format of the sharees API

### PR DESCRIPTION
**Pull request by C0rby**
The sharees API wasn't returning the users and groups arrays correctly.

### CODE REVIEW

File: tests/acceptance/expected-failures-on-OCIS-storage.md [hunk 1]
```diff
@@ -335,14 +335,9 @@ File and sync features in a shared scenario
 -   [apiWebdavEtagPropagation1/moveFileFolder.feature:315](https://github.com/owncloud/core/blob/master/tests/acceptance/features/apiWebdavEtagPropagation1/moveFileFolder.feature#L315)
 
 #### [Searching sharee with displayname](https://github.com/owncloud/ocis/issues/547)
--   [apiSharees/sharees.feature:32](https://github.com/owncloud/core/blob/master/tests/acceptance/features/apiSharees/sharees.feature#L32)
--   [apiSharees/sharees.feature:33](https://github.com/owncloud/core/blob/master/tests/acceptance/features/apiSharees/sharees.feature#L33)
--   [apiSharees/sharees.feature:53](https://github.com/owncloud/core/blob/master/tests/acceptance/features/apiSharees/sharees.feature#L53)
--   [apiSharees/sharees.feature:54](https://github.com/owncloud/core/blob/master/tests/acceptance/features/apiSharees/sharees.feature#L54)
+These tests fail because they contain some configuration steps for oc10 specific configuration.
```
Comment (phil-davis) [inline tests/acceptance/expected-failures-on-OCIS-storage.md:338]: I raised issue https://github.com/owncloud/ocis/issues/2354 to look through the remaining test failing test scenarios - see the issue for details.

File: tests/acceptance/expected-failures-on-OCIS-storage.md [hunk 2]
```diff
@@ -365,42 +360,33 @@ File and sync features in a shared scenario
...
--   [apiSharees/sharees.feature:633](https://github.com/owncloud/core/blob/master/tests/acceptance/features/apiSharees/sharees.feature#L633)
--   [apiSharees/sharees.feature:657](https://github.com/owncloud/core/blob/master/tests/acceptance/features/apiSharees/sharees.feature#L657)
--   [apiSharees/sharees.feature:658](https://github.com/owncloud/core/blob/master/tests/acceptance/features/apiSharees/sharees.feature#L658)
 -   [apiSharees/sharees.feature:682](https://github.com/owncloud/core/blob/master/tests/acceptance/features/apiSharees/sharees.feature#L682)
 -   [apiSharees/sharees.feature:683](https://github.com/owncloud/core/blob/master/tests/acceptance/features/apiSharees/sharees.feature#L683)
 -   [apiSharees/sharees.feature:702](https://github.com/owncloud/core/blob/master/tests/acceptance/features/apiSharees/sharees.feature#L702)
 -   [apiSharees/sharees.feature:703](https://github.com/owncloud/core/blob/master/tests/acceptance/features/apiSharees/sharees.feature#L703)
+These tests fail because we currently can't differentiate between exact matches and not exact matches.
```
Comment (phil-davis) [inline tests/acceptance/expected-failures-on-OCIS-storage.md:379]: Is there an issue for this somewhere? If so, we should put a reference to it.

File: tests/acceptance/expected-failures-on-S3NG-storage.md [hunk 3]
```diff
@@ -371,42 +366,33 @@ File and sync features in a shared scenario
...
--   [apiSharees/sharees.feature:633](https://github.com/owncloud/core/blob/master/tests/acceptance/features/apiSharees/sharees.feature#L633)
--   [apiSharees/sharees.feature:657](https://github.com/owncloud/core/blob/master/tests/acceptance/features/apiSharees/sharees.feature#L657)
--   [apiSharees/sharees.feature:658](https://github.com/owncloud/core/blob/master/tests/acceptance/features/apiSharees/sharees.feature#L658)
 -   [apiSharees/sharees.feature:682](https://github.com/owncloud/core/blob/master/tests/acceptance/features/apiSharees/sharees.feature#L682)
 -   [apiSharees/sharees.feature:683](https://github.com/owncloud/core/blob/master/tests/acceptance/features/apiSharees/sharees.feature#L683)
 -   [apiSharees/sharees.feature:702](https://github.com/owncloud/core/blob/master/tests/acceptance/features/apiSharees/sharees.feature#L702)
 -   [apiSharees/sharees.feature:703](https://github.com/owncloud/core/blob/master/tests/acceptance/features/apiSharees/sharees.feature#L703)
+These tests fail because we currently can't differentiate between exact matches and not exact matches.
```
Comment (phil-davis) [inline tests/acceptance/expected-failures-on-S3NG-storage.md:385]: Is there an issue for this somewhere? If so, we should put a reference to it.

**Review by phil-davis — APPROVED** [review]
LGTM - this fixes a lot of test scenarios. We can sort out issue references for the remaining scenarios later.

**Review by butonic — APPROVED** [review]
🥳

**Comment by phil-davis** [comment]
@ishank011 @labkode can you review this and merge if you are happy with the code changes.

Investigating the remaining `apiSharees/sharees.feature` test scenarios that still fail is a separate issue, and I will get people to look into exactly what real issues remain there after this is merged.

**Review by labkode — APPROVED** [review]

**Pull request merged**